Userland/Libraries/LibGUI/UndoStack.cpp

```diff
@@ -78,12 +86,30 @@ void UndoStack::finalize_current_combo()
 
     auto undo_commands_container = make<UndoCommandsContainer>();
     m_stack.prepend(move(undo_commands_container));
+
+    if (m_clean_index.has_value())
+        m_clean_index = m_clean_index.value() + 1;
```

Comment:
I don't understand this, why does the clean index change? It should only change when marking the undo stack clean.